### ISSUE TITLE
Duplicated warnings for unused features

### ISSUE THREAD

**Issue by tthuem** [FeatureIDE/FeatureIDE#402]
Imported the following project and got the same warning for unused features four times: http://git.iti.cs.ovgu.de/dbse-projects/featureide/tree/master/tutorials/2016-SPLC/projects/ImprovedElevator

![zwischenablage-1](https://cloud.githubusercontent.com/assets/5330357/18325179/6d233baa-7540-11e6-905b-0e4e73c58ed0.png)

Also, make sure that the last comma is removed.

**Comment by tthuem** [comment]
After new import, the warning appeared only once but 6(!) times after closing and opening the project.

**Comment by tthuem** [comment]
It is absolutely fine to repeat the analysis, but prior to creation, all markers of that type need to be removed. That might already fix the problem.

**Comment by tthuem** [comment]
Also change it to be an info and not a warning.

**Comment by tthuem** [comment]
After closing and opening a project, the FeatureIDE icon is missing and the marker for current configurations:

![zwischenablage-1](https://cloud.githubusercontent.com/assets/5330357/18542604/0ed74270-7b2c-11e6-9887-4acd25879500.png)

Alex, please check whether this was introduced by your fix. If not, we create another issue.

**Comment by AlexanderKnueppel** [comment]
Seems to happen becourse of my fix. Oddly enough, it is only the first project that is closed and reopened, which is affected by that, right?

**Comment by tthuem** [comment]
It might have been the first, but later I had several projects with the same problem. It could not even be fixed by importing the project again.

**Comment by tthuem** [comment]
Restarting Eclipse did help.

**Comment by tthuem** [comment]
To sum up: opening and importing projects does not work anymore as expected.

**Comment by AlexanderKnueppel** [comment]
Now, the warnings are gone, I'll try to fix that.

**Comment by AlexanderKnueppel** [comment]
Check is now only performed if a configuration is changed.

**Comment by tthuem** [comment]
Should also be performed when the feature model changes. (Ideally also when a configuration is removed.)

**Comment by AlexanderKnueppel** [comment]
A feature model change leads also to a check. Deleting a configuration does not, if I see it correctly.

**Comment by AlexanderKnueppel** [comment]
The problem is that a project instance is created either at eclipse start, after reopening a closed project, or after importing a project. Every time though the project instance registers itself as a ResourceChangeListener, all pointing to one folder structure/physical project. They are only disposed, when eclipse exits, not when a project is deleted or closed. 

The fix right now removes the project as a listener if it is either closed or deleted. The consequence is that checks are not performed at opening or importing projects (like it is when eclipse starts).  

Solution 1: trigger checks at project construction (but this would trigger checks at eclipse start of each project; could possibly be prevented by some conditions/flag)

Solution 2: ...? Adding a listener only if it does not exist is not an option I guess, as physical files and folders may change but the listeneners are depending on their virtual instantiation. I removed a project from disk where I changed a feature name and imported the original project again. Then I got a message that the renamed feature is not used in any configuration.

Maybe someone with more experience with the feature project construction, configuration checks, and the depending listeners can have a look at this.

**Comment by tthuem** [comment]
Alex, can you summarize what is not working right now?

**Comment by AlexanderKnueppel** [comment]
Check and generating markers is only triggered when a configuration file or the feature model changes. NOT when a project is reopened or imported, or when a configuration file is deleted.

**Comment by tthuem** [comment]
Solution 1 sounds fine to me for now. We have numerous more complex computations anyway.

**Comment by tthuem** [comment]
Closed as the actual problem is fixed. See #424 for the open issues.
